### PR TITLE
ctapipe-fileinfo

### PR DESCRIPTION
**Pull request by kosack**
Just a small fast command-line utility to display information about ctapipe output files. 

- the output is all reference headers in YAML format  (hierarchical by default, but a `--flat` option is available)
- you can optionally write an index table for all files with all headers as columns (uses pandas DataFrame's ability to convert a list of dicts into a table, and then uses astropy.table to write it to any format you want, e.g. `files.fits` or `files.ecsv`)
- files not in hdf5 format are skipped (with a message)
- incomplete hdf5 files will return `{}` (no header)
- you can colorize the output by piping it through `pygmentize -l yaml`

### Help: 
```
% ctapipe-fileinfo --help                                                                                        

usage: ctapipe-fileinfo [-h] [--output-table OUTPUT_TABLE] [--flat] FILENAME [FILENAME ...]

    Display information about ctapipe output files (DL1 or DL2 in HDF5 format).
    Optionally create an index table from all headers


positional arguments:
  FILENAME              filenames of files in ctapipe format

optional arguments:
  -h, --help            show this help message and exit
  --output-table OUTPUT_TABLE
                        generate output in tabular format (default: None)
  --flat                show flat header hierarchy (default: False)
```

### Example 2:

```
%  ctapipe-fileinfo *.DL1.h5

proton_20deg_0deg_run153___cta-prod5-paranal_desert-2147m-Paranal-dark_merged.DL1.h5:
    CTA:
        ACTIVITY:
            ID: eb09130c-1a4e-4a0a-96a0-2607ebca70b1
            NAME: ctapipe-merge
            SOFTWARE:
                NAME: ctapipe
                VERSION: 0.10.5
            START:
                TIME: '2021-03-15 15:30:01.948'
            TYPE: software
        CONTACT:
            EMAIL: ''
            NAME: ''
            ORGANIZATION: CTA Consortium
        INSTRUMENT:
            CLASS: Subarray
            ID: MonteCarloArray
            SITE: Other
            SUBTYPE: unspecified
            TYPE: unknown
            VERSION: unknown
        PROCESS:
            ID: '0'
            SUBTYPE: ''
            TYPE: Simulation
        PRODUCT:
            CREATION:
                TIME: '2021-03-15 15:30:57.161'
            DATA:
                ASSOCIATION: Subarray
                CATEGORY: S
                LEVEL: DL1
                MODEL:
                    NAME: ASWG DL1
                    URL: ''
                    VERSION: v1.1.0
            DESCRIPTION: Merged DL1 Data Product
            FORMAT: hdf5
            ID: 98f44cd6-f642-4af0-aa55-7da1621f0c5e
        REFERENCE:
            VERSION: '1'
```

### Example of table output:
```
ctapipe-fileinfo --output-table index.fits
```
image

### CODE REVIEW

**Comment by kosack** [comment]
Note that this also lets us collect some stats on processing time, for example:

image

File: ctapipe/tools/tests/test_tools.py [hunk 1]
```diff
@@ -285,6 +287,28 @@ def test_info():
...
 
 
+def test_fileinfo(tmp_path, dl1_image_file):
+    """ check we can run ctapipe-fileinfo and get results """
+    import yaml
+    from astropy.table import Table
+
+    index_file = tmp_path / "index.fits"
+    command = f"ctapipe-fileinfo {dl1_image_file} --output-table {index_file}"
+    stream = os.popen(command)
```
Comment (maxnoe) [inline ctapipe/tools/tests/test_tools.py:297]: Better use `ret = subprocess.run`, the os commands are low-level and discouraged for normal use.

Comment (kosack) [reply ctapipe/tools/tests/test_tools.py]: yes good idea. Done

File: ctapipe/tools/fileinfo.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+"""
+
+from pathlib import Path
+import tables
+import yaml
+from astropy.table import Table
+from ctapipe.tools.utils import get_parser
+
+
+def unflatten(dictionary, separator=" "):
```
Comment (maxnoe) [inline ctapipe/tools/fileinfo.py:12]: Is this useful outside this tool? Maybe directly add this to `ctapipe.io.metadata`?

**Review by maxnoe — APPROVED** [review]

**Review by HealthyPear — APPROVED** [review]

**Pull request merged**